mikutsky

mikutsky opened the pull request:
Микуцкий Дмитрий, HW1
Сделал все задания, текст заданий добавил в комментарии. Все доработки могу сделать в течении дня.
По своей инициативе добавил файл "report.js" и накатал туда функции, которые выводят, в удобном виде, результаты тестов функций из домашнего задания. Было интересно разобраться с анонимными функциями, и с прочими неочивидными для меня прежде фичами. По этому и решил добавить к домашке этот файл. Если не нужно делать так - грохну его)
Вопросы, касаются оформления:
1. Как оформлять, какие комментарии нужно писать, а какие нет.
2. Меня немного раздражает когда команда распологается на нескольких строчках, но Prettier по умолчанию разбирает строки более 80 символов. Это критично и нужно привыкнуть или можно настроить что бы не разбивал? А также табуляция которая заменяется пробелами, хотя в VS Code я это настроил, но Prettier упорно меняет на пробелы, как и кавычки пустык строк и некоторые другие мелочи.
3. Мне кажется проверка на корректность переданных в функцию аргументов у меня немного великовата, это нормально или нужно упростить как-то?
4. Писать {} в местах где всего один оператор в них включен я не стал. Так компактней и наглядней ИМХО. Но если нужно везде допишу.

Все что нужно исправить по самому коду жду от Вас. Спасибо!

ЗЫ. Вопросы по оформлению отпали, когда заметил гайд по оформлению на Piazza. Вижу где ошибки, завтра поправлю оформление. Может еще замечания будут.

mikutsky committed:
mikutsky

Микуцкий Дмитрий, HW1
Сделал все задания, текст заданий добавил в комментарии. Все доработки могу сделать в течении дня.
По своей инициативе добавил файл "report.js" и накатал туда функции, которые выводят, в удобном виде, результаты тестов функций из домашнего задания. Было интересно разобраться с анонимными функциями, и с прочими неочивидными для меня прежде фичами. По этому и решил добавить к домашке этот файл. Если не нужно делать так - грохну его)

mikutsky commented:
> Микуцкий Дмитрий, HW1
> Сделал все задания, текст заданий добавил в комментарии. Все доработки могу сделать в течении дня.
> По своей инициативе добавил файл "report.js" и накатал туда функции, которые выводят, в удобном виде, результаты тестов функций из домашнего задания. Было интересно разобраться с анонимными функциями, и с прочими неочивидными для меня прежде фичами. По этому и решил добавить к домашке этот файл. Если не нужно делать так - грохну его)
> Вопросы, касаются оформления:
> 
> 1. Как оформлять, какие комментарии нужно писать, а какие нет.
> 2. Меня немного раздражает когда команда распологается на нескольких строчках, но Prettier по умолчанию разбирает строки более 80 символов. Это критично и нужно привыкнуть или можно настроить что бы не разбивал? А также табуляция которая заменяется пробелами, хотя в VS Code я это настроил, но Prettier упорно меняет на пробелы, как и кавычки пустык строк и некоторые другие мелочи.
> 3. Мне кажется проверка на корректность переданных в функцию аргументов у меня немного великовата, это нормально или нужно упростить как-то?
> 4. Писать {} в местах где всего один оператор в них включен я не стал. Так компактней и наглядней ИМХО. Но если нужно везде допишу.
> 
> Все что нужно исправить по самому коду жду от Вас. Спасибо!
> 
> ЗЫ. Вопросы по оформлению отпали, когда заметил гайд по оформлению на Piazza. Вижу где ошибки, завтра поправлю оформление. Может еще замечания будут.

mikutsky committed:
Update app.js

Поправил текст сообщения об ошибке в функциях getArray() и doubleArray()

nkipen commented on app.js:
```diff
@@ -0,0 +1,196 @@
+//------------------------------------------------------------
+//Создать функцию multiply, которая будет принимать любое количество чисел и возвращать их произведение:
+//multiply(1,2,3) = 6 (1*2*3). Если нет ни одного аргумента, вернуть ноль: multiply(); //0
+function multiply(args = 1) {
```
что за аргумент args? не совсем поняла для чего он нужен здесь, и почему по дефолту равен 1?

mikutsky replied:
Дефоултное значение осталось с прошлой версии функции, забыл удалить. args - оказывается тоже не нужен. Я думал если я ожидаю какой-то аргумент, то нужно указывать его в скобках. Но теперь когда ты спростила, попробовал - оно и без этого работает. Т.е. выходит если я перебираю все аргументы через arguments, то при объявлении функции можно описание аргумента опустить? ...класс))
Исправлю. Эта ошибка в задании №1 и №7. Спасибо!

nkipen replied:
да, если ты не ожидаешь конкретное значение, его можно при обьявлении фун-ии не задавать.

arguments всегда есть у функции, это ее коллекция аргументов. тебе конкретоно в этой задаче нужны только arguments, поэтому args не нужен)

nkipen commented:
**комментраии**
работа хорошая, только очень много комментов, ну ты и сам уже, как я увидела, понял)
сейчас суперпростые задачи, для них не нужна такая грубая проверка и обработка ошибок, и тем более отдельный файл для вывоа. достаточно при проверке на тот же undefined/пустоту возвращать false и на месте консолить результат выполнения функции.

**по втоим вопросам, кроме тех, которые сами отпали:**
2. чекну и скажу
3. проверка строгая, просто лишняя работа)
4. можно без скобок, главное придерживайся по проекту этого принципа.

**к исправлению**
файлик с выводом и обработку ошибок не трогай, а комменты убери. но в след работах немного попроще делай, а то у меня двоится в глазах после проверки :DDDDDD
+ у тебя условия\комментрарии уходят далеко вправо и появляется скролл. установи себе ширину строки в редакторе и переноси (или настрой авто-перенос, я помогу если не разберешься). должно быть 80-120 символов в строке (80 хватает с головой).

mikutsky committed:
Update app.js

1. Лишние коменты убрал.
2. Все строки уменьшил до 80 символов
ЗЫ с проверкой аргументов понял, буду в будущем проще и прямолинейней в этом деле)

mikutsky committed:
Merge branch 'homework' of https://github.com/mikutsky/HW1 into homework

mikutsky committed:
Update app.js

Final changes:
Fixed code view. All indents are the same - spaces.
